modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/KnowledgeBaseArticleResourceImpl.java

```diff
@@ -376,6 +401,73 @@ public void putKnowledgeBaseArticleUnsubscribe(Long knowledgeBaseArticleId)
 		_kbArticleService.unsubscribeKBArticle(knowledgeBaseArticleId);
 	}
 
+	@Override
+	public KnowledgeBaseArticle
+			putSiteKnowledgeBaseArticleByExternalReferenceCode(
+				Long siteId, String externalReferenceCode,
+				KnowledgeBaseArticle knowledgeBaseArticle)
+		throws Exception {
+
+		KBArticle kbArticle =
+			_kbArticleLocalService.fetchLatestKBArticleByExternalReferenceCode(
+				siteId, externalReferenceCode);
+
+		Long parentKnowledgeBaseArticleId =
+			knowledgeBaseArticle.getParentKnowledgeBaseArticleId();
+
+		if (parentKnowledgeBaseArticleId == null) {
+			parentKnowledgeBaseArticleId =
+				KBArticleConstants.DEFAULT_PARENT_RESOURCE_PRIM_KEY;
+		}
+
+		if (kbArticle == null) {
+			return _toKnowledgeBaseArticle(
+				_kbArticleService.addKBArticle(
```

Comment:
Is this duplicated somewhere?

Reply:
Reviewing the code, I saw it could be duplicated in the private method `_getKnowledgeBaseArticle` (a strange name for me for a method that adds a KB). 
Anyway, maybe I can extract it (and the update) to a unique method.